test/integration/clustercidr/ipam_test.go

```diff
@@ -110,22 +112,386 @@ func TestIPAMMultiCIDRRangeAllocatorType(t *testing.T) {
+
+	_, kubeConfig, tearDownFn := framework.StartTestServer(t, framework.TestServerSetup{
+		ModifyServerRunOptions: func(opts *options.ServerRunOptions) {
+			// Disable ServiceAccount admission plugin as we don't have serviceaccount controller running.
+			opts.Admission.GenericAdmission.DisablePlugins = []string{"ServiceAccount", "TaintNodesByCondition"}
+			opts.APIEnablement.RuntimeConfig.Set("networking.k8s.io/v1alpha1=true")
+		},
+	})
+	defer tearDownFn()
+
+	clientSet := clientset.NewForConfigOrDie(kubeConfig)
+	sharedInformer := informers.NewSharedInformerFactory(clientSet, 1*time.Hour)
+
+	ipamController := booststrapMultiCIDRRangeAllocator(t, clientSet, sharedInformer)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go ipamController.Run(ctx.Done())
+	sharedInformer.Start(ctx.Done())
+
+	t.Run("delete cc with node associated", func(t *testing.T) {
+
+		// Create a ClusterCIDR.
+		clusterCIDR := makeClusterCIDR("dualstack-cc-del", "192.168.0.0/23", "fd00:30:100::/119", 8, nodeSelector(map[string][]string{"ipv4": {"true"}, "ipv6": {"true"}}))
+		if _, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Create(context.TODO(), clusterCIDR, metav1.CreateOptions{}); err != nil {
+			t.Fatal(err)
+		}
+
+		// Sleep for one second to make sure the controller processes the newly created ClusterCIDR.
+		time.Sleep(1 * time.Second)
+
+		// Create a node, which gets pod CIDR from the clusterCIDR created above.
+		node := makeNode("dualstack-node", map[string]string{"ipv4": "true", "ipv6": "true"})
+		expectedPodCIDRs := []string{"192.168.0.0/24", "fd00:30:100::/120"}
+		if _, err := clientSet.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{}); err != nil {
+			t.Fatal(err)
+		}
+		if gotPodCIDRs, err := nodePodCIDRs(clientSet, node.Name); err != nil {
+			t.Fatal(err)
+		} else if !reflect.DeepEqual(expectedPodCIDRs, gotPodCIDRs) {
+			t.Errorf("unexpected result, expected Pod CIDRs %v but got %v", expectedPodCIDRs, gotPodCIDRs)
+		}
+
+		// Delete the ClusterCIDR
+		if err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Delete(context.TODO(), clusterCIDR.Name, metav1.DeleteOptions{}); err != nil {
+			t.Fatal(err)
+		}
+
+		// Sleep for five seconds to make sure the ClusterCIDR exists with a deletion timestamp after marked for deletion.
+		time.Sleep(5 * time.Second)
+
+		//Wait and make sure that the ClusterCIDR is not deleted, as there is a node associated with it.
+		if err := wait.PollImmediate(time.Second, wait.ForeverTestTimeout, func() (bool, error) {
+			cc, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Get(context.TODO(), clusterCIDR.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+
+			if cc != nil && !cc.DeletionTimestamp.IsZero() {
+				return true, nil
+			}
+			return false, fmt.Errorf("clustercidr was deleted or deletion timestamp was not set")
```

Comment:
if you send an error  you exit immediately, you can just log the message and return false, nil

Reply:
but thinking about this more, you don't need the `wait.PollImmediate`, just check 

```
		cc, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Get(context.TODO(), clusterCIDR.Name, metav1.GetOptions{})
			if err != nil {
				t.Fatal(err)
			}

			if cc == nil {
				t.Fatalf("expected Cluster CIDR got nil")
			}
                         if cc.DeletionTimestamp.IsZero() {
                            t.Fatalf("expected Cluster CIDR to have set a deletion timestamp ")
                         }
```

Reply:
Ack, updated